code/surface_area.py

```diff
@@ -0,0 +1,421 @@
+                    print(f"✗ Minimum: {data_min:.6f} (should be non-negative)")
+
+                if data_max > 0:
+                    validation_results["Maximum"] = True
+                    print(f"✓ Maximum: {data_max:.6f} (valid positive value)")
+                else:
+                    validation_results["Maximum"] = False
+                    print(f"✗ Maximum: {data_max:.6f} (should be positive)")
+
+                if data_mean > 0:
+                    validation_results["Mean"] = True
+                    print(f"✓ Mean: {data_mean:.6f} (valid positive value)")
+                else:
+                    validation_results["Mean"] = False
+                    print(f"✗ Mean: {data_mean:.6f} (should be positive)")
+
+            else:
+                validation_results["Number of Vertices"] = False
+                validation_results["Minimum"] = False
+                validation_results["Maximum"] = False
+                validation_results["Mean"] = False
+                print("✗ No data found in the metric file")
+
+        else:
+            validation_results["Maps to Surface"] = False
+            validation_results["Map Name"] = False
+            validation_results["Number of Vertices"] = False
+            validation_results["Minimum"] = False
+            validation_results["Maximum"] = False
+            validation_results["Mean"] = False
+            print("✗ Maps to Surface: false (no data arrays found)")
+
+        # Overall validation result
+        all_passed = all(validation_results.values())
+
+        if all_passed:
+            print("✓ All field validations passed")
+        else:
+            failed_checks = [k for k, v in validation_results.items() if not v]
+            print(f"✗ Failed validations: {', '.join(failed_checks)}")
+
+        return all_passed
+
+    except Exception as e:
+        print(f"✗ Validation failed: {e}")
+        return False
+
+
+def test_output_file(output_metric: Path, validate: bool = True):
+    """
+    Test if the output metric file exists and has valid data fields.
+    """
+    # First run the niwrap file_information to ensure it works
+    try:
+        wb.file_information(str(output_metric))
+        print("✓ niwrap file information executed successfully")
+    except Exception as e:
+        print(f"✗ niwrap file information failed: {e}")
+        return False
+
+    if validate:
+        # Now validate the actual data fields
+        is_valid = validate_output_file_data(output_metric)
+
+        if is_valid:
+            print("✓ Complete output file validation successful")
+        else:
+            print("✗ Output file validation failed")
+
+        return is_valid
+
+
+def find_surface_files(
+    input_dir: Path, patterns: Iterable[str] | None = None
+) -> list[Path]:
+    """
+    Find surface files matching the specified patterns.
+    """
+    if patterns is None:
+        patterns = ["*mid*.surf.gii"]
```

Comment:
I think one of the templates might use a `rsl.gii` extension instead of `surf.gii`, so could also add that pattern here.